vispy/color/colormap.py

```diff
@@ -427,18 +431,13 @@ def map(self, x):
         return self._map_function(self.colors.rgba, x, self._controls)
 
     def texture_lut(self):
-        """Return a texture2D object for LUT after its value is set."""
-        if self.texture_map_data is not None:
-            interpolation_mode = 'linear' \
-                if(str(self.interpolation) == 'linear') \
-                else 'nearest'
-            texture_LUT = \
-                vispy.gloo.Texture2D(np.zeros(self.texture_map_data.shape),
-                                     interpolation=interpolation_mode)
-            texture_LUT.set_data(self.texture_map_data,
-                                 offset=None, copy=True)
-        else:
-            texture_LUT = None
+        """Return a texture2D object for LUT after its value is set. Can be None."""
+        if self.texture_map_data is None:
+            return None
+        interp = 'linear' if (str(self.interpolation) == 'linear') else 'nearest'
```

Comment:
```suggestion
        interp = 'linear' if self.interpolation == 'linear' else 'nearest'
```

I know this isn't your code, but interpolation should never be not a string. Since the main point of this PR is cleanup, let's go one more step.